### PR TITLE
Add documentation for volume group snapshot

### PR DESCRIPTION
**Pull request by leonardoce**
```release-notes
Improve the documentation to cover Volume Group Snapshots.
```

### CODE REVIEW

File: book/src/group-snapshot.md [hunk 1]
```diff
@@ -0,0 +1,165 @@
+# Volume Group Snapshot Feature
+
+## Status
+
+Status | Min K8s Version | Max K8s Version | snapshot-controller Version | snapshot-validation-webhook Version | CSI external-snapshotter sidecar Version | external-provisioner Version
+--|--|--|--|--|--|--
+Alpha  | 1.27 | - | 7.0+ | 7.0+ | 7.0+ | 3.5+
```
Comment (xing-yang) [inline book/src/group-snapshot.md:7]: external-provisioner version 3.5+ -> 4.0+

File: book/src/group-snapshot.md [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+responsible for watching the VolumeGroupSnapshot CRD objects and manges the
+creation and deletion lifecycle of volume group snapshots.
+
+The CSI external-snapshotter sidecar watches Kubernetes
+VolumeGroupSnapshotContent CRD objects and triggers
+CreateVolumeGroupSnapshot/DeleteVolumeGroupSnapshot against a CSI endpoint.
+
+## Snapshot Validation Webhook
+
+The validating webhook server is updated to validate volume group snapshot
```
Comment (xing-yang) [inline book/src/group-snapshot.md:91]: Can you add a note similar to what we have in the release note that the webhook is deprecated?
https://github.com/kubernetes-csi/external-snapshotter/releases/tag/client%2Fv8.0.0

File: book/src/group-snapshot.md [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+Create a _PVC_ from a _VolumeSnapshot_ that is part of the group snapshot:
+```
+# In the command below, the volume snapshot name should be chosen from
+# the ones listed in the output of the previous command
+sed 's/new-snapshot-demo-v1/snapshot-4bcc4a322a473abf32babe3df5779d14349542b1f0eb6f9dab0466a85c59cd42-2024-06-19-12.35.17/' restore.yaml | kubectl create -f -
+```
+
+## Examples
+
+See the [Drivers](drivers.md) for a list of CSI drivers that implement the snapshot feature.
```
Comment (Madhu-1) [inline book/src/group-snapshot.md:165]: do we need to change `snapshot feature` to `group snapshot feature`?

**Comment by leonardoce** [comment]
I moved the group snapshot feature page inside the `data sources` section, where the snapshot feature is.
I should have addressed @xing-yang and @Madhu-1's comments too.
Thank you!

File: book/src/SUMMARY.md [hunk 4]
```diff
@@ -33,6 +33,7 @@
         - [Data Sources](volume-datasources.md)
             - [Cloning](volume-cloning.md)
             - [Volume Snapshot & Restore](snapshot-restore-feature.md)
+            - [Volume Group Snapshot & Restore](group-snapshot.md)
```
Comment (Madhu-1) [inline book/src/SUMMARY.md:36]: should we name this file to `grou-snapshot-restore-feature.md` to keep it similar to `volume snapshot`?

File: book/src/group-snapshot.md [hunk 5]
```diff
@@ -0,0 +1,167 @@
+# Volume Group Snapshot Feature
+
+## Status
+
+Status | Min K8s Version | Max K8s Version | snapshot-controller Version | snapshot-validation-webhook Version | CSI external-snapshotter sidecar Version | external-provisioner Version
+--|--|--|--|--|--|--
+Alpha  | 1.27 | - | 7.0+ | 7.0+ | 7.0+ | 4.0+
+
+**IMPORTANT**: The validation logic for VolumeGroupSnapshots and VolumeGroupSnapshotContents has been replaced by CEL validation rules. The validating webhook is now only being used for VolumeGroupSnapshotClasses to ensure that there's at most one class per CSI Driver. The validation webhook is deprecated and will be removed in the next release
```
Comment (Madhu-1) [inline book/src/group-snapshot.md:9]: can we update this to mention only one **default** class per driver is allowed?

File: book/src/snapshot-restore-feature.md [hunk 6]
```diff
@@ -8,6 +8,8 @@ Alpha | 1.12 | 1.12 | | | 0.4.0 <= version < 1.0 | 0.4.1 <= version < 1.0
 Alpha | 1.13 | 1.16 | | | 1.0.1 <= version < 2.0 | 1.0.1 <= version < 1.5
 Beta  | 1.17 | - | 2.0+ | 3.0+ | 2.0+ | 1.5+
 
+**IMPORTANT**: The validation logic for VolumeSnapshots and VolumeSnapshotContents has been replaced by CEL validation rules. The validating webhook is now only being used for VolumeSnapshotClasses to ensure that there's at most one class per CSI Driver. The validation webhook is deprecated and will be removed in the next release.
```
Comment (Madhu-1) [inline book/src/snapshot-restore-feature.md:11]: same here as well can we mention about default class

**Review by Madhu-1 — COMMENTED** [review]
small suggestions, @leonardoce can you please also add release-notes?

**Comment by leonardoce** [comment]
Thank you @Madhu-1. I added the release notes, too.

**Comment by Madhu-1** [comment]
/lgtm
/assign @xing-yang 

@leonardoce Thank you

**Pull request merged**